starlette/_crypto.py

```diff
@@ -0,0 +1,14 @@
+import hashlib
+import inspect
+
+
+# TODO: Remove when dropping support for PY38. inspect.signature() is used to
+# detect whether the usedforsecurity argument is available as this fix may also
+# have been applied by downstream package maintainers to other versions in
+# their repositories.
+if "usedforsecurity" in inspect.signature(hashlib.md5).parameters:
+    def md5_hexdigest(data: bytes, *, usedforsecurity: bool = True) -> str:
```

Comment:
Do we straight up want to raise an error if `usedforsecurity=True` in either of these implementations? Seems like overkill but it'd also be easy to implement.

Reply:
Why would we want to do that? On which scenario?

Reply:
We currently don't have any scenarios in which we'd want to pass `usedforsecurity=True` (and arguably md5 should never be used for security). Adding an error would just make it very explicit and add a barrier to misusing md5 for security in future changes.

Reply:
I think there are 2 scenarios this can help with:
1. Keeps us from shooting ourselves in the foot in the future once this PR is long forgotten.
1. Keeps users from misusing this function down the road (the module is private by name, but this is Python so someone could still use it).

Reply:
No, let's just make the minimum possible change here.